theano/tensor/nnet/ctc.py

```diff
@@ -239,5 +244,6 @@ def local_ctc_no_grad(node):
     if isinstance(node.op, ConnectionistTemporalClassification):
         if len(node.outputs) > 1:
             if len(node.outputs[1].clients) == 0:   # gradient is not used
-                return [ConnectionistTemporalClassification(compute_grad=False)(*node.inputs), None]
+                activations, labels, input_lengths, _ = node.inputs
+                return [ConnectionistTemporalClassification(compute_grad=False)(activations, labels, input_lengths), None]
```

Comment:
This would be a good reason why the current make_node isn't good. We need the old code to work. Otherwise, the cloning system won't work.

A quick work around would be to add an openmp_enabled parameter to make_node() and have it default to None. When none, it should use `self.openmp`. But do we really support having the openmp setting change for the same node in the same Theano function? I don't think we should support that.

So that point more that we should use params for that.

Reply:
Indeed, I don't think this kind of change should be supported too, I'll also fix it.